injector/src/main/java/edu/ucr/cs/riple/injector/modifications/Deletion.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022 Nima Karimipour
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+package edu.ucr.cs.riple.injector.modifications;
+
+import com.github.javaparser.Position;
+import java.util.List;
+import javax.lang.model.element.ElementKind;
+
+/** Represents a deletion of a content in the source file. */
+public class Deletion extends Modification {
+
+  /** The end position of content which should be removed. */
+  private final Position endPosition;
+
+  public Deletion(String content, Position startPosition, Position endPosition, ElementKind kind) {
+    super(content, startPosition, kind);
+    this.endPosition = new Position(endPosition.line - 1, endPosition.column - 1);
+  }
+
+  @Override
+  public void visit(List<String> lines) {
+    int start = startPosition.column;
+    int end = endPosition.column;
+    StringBuilder line = new StringBuilder(lines.get(startPosition.line));
```

Comment:
This assumes `startPosition.line == endPosition.line` for deletions, no? Meaning that the entire annotation should be within the same line.

1. We probably should make the assumption explicit in comments and maybe a preconditions check
2. This might get broken for the case of `@SuppressWarnings` annotations, eg.

```
@SuppressWarnings(
    "NullAway.Init")
@SomeOtherAnnotation(...)
private Object f1;
```

Reply:
Yes, we do not support deletion of annotations written in multiple lines. It wasn't also possible to make such requests from Injector, as we can only make `RemoveAnnotation` instance for marker annotations. This check was implicitly applied, however, I added the precondition explicitly 970f82ed397473080751e5d8051f769645b72f53.